api/src/server.ts

```diff
@@ -1,84 +1,26 @@
-import dotenv from 'dotenv';
-import { destroyDBConnections } from './client/db';
-import express from './express';
-import { getSequelize, closeSequelize } from './orm';
-import { initialModels, Task, User } from './models';
-
-dotenv.config();
-
-const isValidEnvironment = () => {
-  const {
-    PORT,
-    DB_HOST,
-    DB_PORT,
-    DB_NAME,
-    DB_USERNAME,
-    DB_PASSWORD,
-    DB_SCHEMA,
-    JWT_SECRET_KEY,
-    TOKEN_HEADER_KEY,
-    JWT_EXPIRES_IN,
-  } = process.env;
-  if (
-    !PORT ||
-    !DB_HOST ||
-    !DB_PORT ||
-    !DB_NAME ||
-    !DB_USERNAME ||
-    !DB_PASSWORD ||
-    !DB_SCHEMA ||
-    !JWT_SECRET_KEY ||
-    !TOKEN_HEADER_KEY ||
-    !JWT_EXPIRES_IN
-  ) {
-    return false;
-  }
-};
-
-const main = async () => {
-  const error = `This service will be terminated.
-  Check the environment, some environment variables are missing.`;
-  const isValid = isValidEnvironment();
-  if (isValid) {
-    console.error(error);
-    process.exit(1);
-  }
-
-  const sequelize = await getSequelize();
-  if (!sequelize) {
-    console.error(error);
-    process.exit(1);
+require('dotenv').config();
+const express = require('express');
+const sequelize = require('./orm');
+const cors = require('cors')
+const router = require('./routes/index')
+
+const PORT = process.env.PORT || 3000;
+
+const app = express();
+app.use(cors())
+app.use(express.json())
+app.use('/v1', router)
+
+const start = async () => {
+  try {
+    await sequelize.authenticate() // connect DB
+    await sequelize.sync() // this function checks the state of the database against the data schema
+    app.listen(PORT, () => {
+      console.log(`Server started on port: ${PORT}`);
+    });
+  } catch (e) {
+    console.log(e);
   }
-
-  await initialModels(sequelize);
-
-  // Test the models if you like
-  // const allUsers = await User.findAll();
-  // const allTasks = await Task.findAll();
-  // console.log(allUsers);
-  // console.log(allTasks);
-
-  const port = process.env.PORT;
-
-  express.listen(port, () => {
-    console.log(`[server]: Server is running at http://localhost:${port}`);
-  });
-
-  const shutdown = async (signal: string) => {
-    await destroyDBConnections();
-    await closeSequelize();
-
-    console.info(`[${signal}] This service is shutting down!`);
-    process.exit(0);
-  };
-
-  process.on('SIGTERM', async () => {
-    await shutdown('SIGTERM');
-  });
-
-  process.on('SIGINT', async () => {
-    await shutdown('SIGINT');
-  });
```

Comment:
I've been contemplating the idea of gracefully shutting down all third-party connections by listening to [SIGTERM](https://www.gnu.org/software/libc/manual/html_node/Termination-Signals.html#index-SIGTERM) (happens when OS kill process) and [SIGINT](https://www.gnu.org/software/libc/manual/html_node/Termination-Signals.html#index-SIGTERM) (happens when Control + C) signals before we halt our service. I believe this approach could enhance our system's robustness and reliability.